ppcs/ppcTODO-signature-named-parameters.md

```diff
@@ -0,0 +1,201 @@
+
+```perl
+sub new_unix ( $class, %args )
+{
+    my $path   = $args{path};
+    my $listen = $args{listen} // 5;
+    ...
+}
+```
+
+Already the new code is shorter and more consise. Additionally it contains error checking that complains about missing mandatory keys, or unrecognised keys, which the previous version of the code did not include.
+
+Another example, this time from [`Text::Treesitter::QueryCursor`](https://metacpan.org/pod/Text::Treesitter::QueryCursor). This method takes an optional argument named `multi`, tested for truth. If absent it should default to false.
+
+```perl
+sub next_match_captures ( $self, :$multi = 0 )
+{
+    ...
+}
+```
+
+The previous version of this code did include complaints about unrecognised keys, and was rather longer because of it:
+
+```perl
+sub next_match_captures ( $self, %options )
+{
+    my $multi = delete $options{multi};
+    keys %options and
+        croak "Unrecognised options to ->next_captures: " . join( ", ", keys %options );
+
+    ...
+}
+```
+
+## Prototype Implementation
+
+The [`XS::Parse::Sublike`](https://metacpan.org/pod/XS::Parse::Sublike) module contains parsing to allow third-party syntax modules to parse subroutine-like constructions, and includes a parser for named parameters already having this syntax. These are also made available to regular subroutines via the `extended` keyword provided by [`Sublike::Extended`](https://metacpan.org/pod/Sublike::Extended).
+
+Additionally, other existing CPAN modules already parse syntax in this, or a very similar format:
+
+* [`Function::Parameters`](https://metacpan.org/pod/Function::Parameters)
+
+* [`Kavorka`](https://metacpan.org/dist/Kavorka/view/lib/Kavorka/Manual/Signatures.pod)
+
+* [`Method::Signatures`](https://metacpan.org/pod/Method::Signatures)
+
+All of these use the leading-colon syntax in a signature declaration to provide named parameters in the same style as this proposal. It would appear we are in good company here.
+
+## Future Scope
+
+* If the Metaprogramming API (PPC0022) gains introspection abilities to enquire about subroutine signature parameters, further consideration will need to be made in that API on how to represent the extra kinds of parameters added by this specification.
+
+## Rejected Ideas
+
+* This specification intentionally makes no changes to the call-site of any subroutines using named parameters.
+
+## Open Issues
+
+### Parameter Names
+
+How to specify a named parameter whose name is anything other than the name of the lexical variable into which its value is assigned? This is more of note when considering that traditional argument handling techniques involving assignment into hashes can handle more styles of name that would be invalid for lexical variables - for example, names including hyphens.
+
+Perhaps a solution would be to permit attributes on parameter variables, then define a `:name` attribute for this purpose.
+
+```perl
+sub display ( $message, :$no_colour :name(no-colour) ) {
+    ...
```

Comment:
The semantics of this aren't defined here, but if this acts as simply an override on the parameter key name rather than an addition to a list of parameter key names, I would like that over my earlier suggestion of not having a colon. :-)